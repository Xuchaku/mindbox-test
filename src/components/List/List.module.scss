.list {
  position: relative;
  -webkit-box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  -moz-box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  background-color: white;
  z-index: 5;
}

.firstShadow {
  position: absolute;
  left: 50%;
  border: 1px solid #ccc;
  z-index: -1;
  bottom: 0;
  transform: translateX(-50%) translateY(100%);
}
.secondShadow {
  position: absolute;
  bottom: -6px;
  -webkit-box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  -moz-box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  box-shadow: 0px -1px 8px 3px rgba(34, 60, 80, 0.2);
  border: 1px solid #ccc;
  left: 50%;
  z-index: -2;
  transform: translateX(-50%) translateY(100%);
}
