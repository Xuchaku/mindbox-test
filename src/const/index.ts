import { Task } from "../types/Task";

export const tabStyle = { border: "1px solid #ccc", borderRadius: "4px" };
export const TODOS_MOCK: Task[] = [
  { completed: false, text: "Тестовое задание", id: 1 },
  { completed: true, text: "Прекрасный код", id: 2 },
  { completed: false, text: "Покрытие тестами", id: 3 },
];
