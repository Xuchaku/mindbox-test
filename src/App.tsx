import { Alert, AlertIcon, Center, Heading, Stack } from "@chakra-ui/react";
import List from "./components/List";
import styles from "./App.module.scss";
import { TODOS_MOCK } from "./const";

function App() {
  return (
    <>
      <Stack spacing={3} w={500}>
        <Alert status="info">
          <AlertIcon />
          Добавление новой задачи осуществляется при вводе в input и нажатии
          Enter
        </Alert>
      </Stack>
      <Center>
        <Heading className={styles.head} fontWeight={"light"} size="4xl">
          todos
        </Heading>
      </Center>
      <List initTodos={TODOS_MOCK} />
    </>
  );
}

export default App;
