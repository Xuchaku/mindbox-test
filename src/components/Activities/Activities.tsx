import {
  Box,
  Button,
  Flex,
  Spacer,
  Tab,
  TabList,
  Tabs,
} from "@chakra-ui/react";
import React, { useState } from "react";
import { tabStyle } from "../../const";
import { Filter } from "../../types/Filter";

import { ActivitiesProps } from "../../types/ActivitiesProps";

export default function Activities({
  todos,
  setFilter,
  handleClear,
}: ActivitiesProps) {
  const [activeIndex, setActiveIndex] = useState(0);

  const tabs: Filter[] = ["All", "Active", "Completed"];
  const itemsLeft = todos.filter((todo) => !todo.completed).length;

  function handleChange(index: number) {
    setActiveIndex(index);
    setFilter(tabs[index]);
  }
  console.log("rerre");

  return (
    <Flex p={4}>
      <Box>
        <Button fontWeight={"light"} colorScheme="brand" variant="ghost">
          {itemsLeft} items left
        </Button>
      </Box>
      <Spacer />
      <Box>
        <Tabs onChange={handleChange} variant={"unstyled"} index={activeIndex}>
          <TabList gap={4}>
            {tabs.map((tab) => {
              return (
                <Tab key={tab} fontWeight={"light"} p={1} _selected={tabStyle}>
                  {tab}
                </Tab>
              );
            })}
          </TabList>
        </Tabs>
      </Box>
      <Spacer />
      <Box>
        <Button
          fontWeight={"light"}
          colorScheme="brand"
          variant="ghost"
          onClick={handleClear}
        >
          Clear completed
        </Button>
      </Box>
    </Flex>
  );
}
