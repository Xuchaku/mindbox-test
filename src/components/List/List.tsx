import { useCallback, useMemo, useState } from "react";
import { Container, VStack } from "@chakra-ui/react";

import { TODOS_MOCK } from "../../const";
import Activities from "../Activities";
import Todo from "../Todo";
import InputAdd from "../InputAdd";

import { Filter } from "../../types/Filter";
import { Task } from "../../types/Task";

import styles from "./List.module.scss";

export default function List() {
  const [todos, setTodos] = useState<Task[]>(TODOS_MOCK);
  const [filter, setFilter] = useState<Filter>("All");

  const memoChangeTodoCheck = useCallback((id: number) => {
    setTodos((prevTodos) => {
      return prevTodos.map((todo) => {
        return todo.id === id ? { ...todo, completed: !todo.completed } : todo;
      });
    });
  }, []);

  const memoAddTodo = useCallback((text: string) => {
    const newTask: Task = { text, id: todos.length + 1, completed: false };
    setTodos([...todos, newTask]);
  }, []);

  function clearTodos() {
    setTodos((prevTodos) => {
      return prevTodos.filter((todo) => !todo.completed);
    });
  }

  const filteredTodos = useMemo(() => {
    if (filter === "All") {
      return todos;
    } else if (filter === "Active") {
      return todos.filter((todo) => !todo.completed);
    } else if (filter === "Completed") {
      return todos.filter((todo) => todo.completed);
    } else {
      return todos;
    }
  }, [todos, filter]);

  return (
    <Container maxW="2xl" className={styles.list} p={0}>
      <InputAdd handleSubmit={memoAddTodo} />
      <VStack align="stretch" spacing={0}>
        {filteredTodos.map((todo) => {
          return (
            <Todo
              key={todo.id}
              todo={todo}
              handleChange={memoChangeTodoCheck}
            />
          );
        })}
      </VStack>
      <Activities
        todos={todos}
        setFilter={setFilter}
        handleClear={clearTodos}
      />
    </Container>
  );
}
