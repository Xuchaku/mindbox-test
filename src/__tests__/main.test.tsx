import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { TODOS_MOCK } from "../const";
import Activities from "../components/Activities";
import App from "../App";
import List from "../components/List";

const onClear = jest.fn();

describe("Activities component", () => {
  it("list renders", () => {
    render(<List initTodos={TODOS_MOCK} />);
    expect(screen.getByText(/Тестовое задание/)).toBeInTheDocument();
    expect(screen.getByText(/Прекрасный код/)).toBeInTheDocument();
    expect(screen.getByText(/Покрытие тестами/)).toBeInTheDocument();
  });

  it("list snapshot", () => {
    const list = render(<List initTodos={TODOS_MOCK} />);
    expect(list).toMatchSnapshot();
  });

  it("Activities renders", () => {
    render(
      <Activities
        todos={TODOS_MOCK}
        handleClear={onClear}
        setFilter={() => {}}
      />
    );

    expect(screen.getByText("All")).toBeInTheDocument();
    expect(screen.getByText("Active")).toBeInTheDocument();
    expect(screen.getByText("Completed")).toBeInTheDocument();
  });
  it("Clear function called", async () => {
    render(
      <Activities
        todos={TODOS_MOCK}
        handleClear={onClear}
        setFilter={() => {}}
      />
    );
    const btn = screen.getByText("Clear completed");
    await userEvent.click(btn);
    expect(onClear).toHaveBeenCalled();
  });
  it("Input works", async () => {
    render(<App />);
    expect(screen.queryByDisplayValue(/Новая задача/)).toBeNull();

    await userEvent.type(screen.getByRole("textbox"), "new");

    expect(screen.getByRole("textbox")).toHaveValue("new");
  });

  it("Add todo works", async () => {
    render(<App />);
    expect(screen.queryByDisplayValue(/Новая задача/)).toBeNull();

    await userEvent.type(screen.getByRole("textbox"), "new{enter}");
    expect(screen.queryByDisplayValue(/new/)).toBeNull();
    expect(screen.getByText("new"));
  });

  it("Clear todos works", async () => {
    render(<App />);
    const todos_completed = TODOS_MOCK.filter((todo) => todo.completed);
    const btn = screen.getByText("Clear completed");
    await userEvent.click(btn);
    todos_completed.forEach((todo) => {
      expect(screen.queryByText(todo.text)).toBeNull();
    });
  });
  it("Tabs works", async () => {
    render(<App />);
    const todos_active = TODOS_MOCK.filter((todo) => !todo.completed);
    const todos_completed = TODOS_MOCK.filter((todo) => todo.completed);

    const tabs = screen.getAllByRole("tab", { selected: false });
    //click Active
    await userEvent.click(tabs[0]);
    todos_active.forEach((todo) => {
      expect(screen.getByText(todo.text)).toBeInTheDocument();
    });
    expect(screen.queryByText(todos_completed[0].text)).toBeNull();

    //click Completed
    await userEvent.click(tabs[1]);
    todos_completed.forEach((todo) => {
      expect(screen.getByText(todo.text)).toBeInTheDocument();
    });
    expect(screen.queryByText(todos_active[0].text)).toBeNull();
    expect(screen.queryByText(todos_active[1].text)).toBeNull();
  });

  it("activities snapshot", () => {
    const list = render(
      <Activities
        todos={TODOS_MOCK}
        handleClear={onClear}
        setFilter={() => {}}
      />
    );
    expect(list).toMatchSnapshot();
  });
});
