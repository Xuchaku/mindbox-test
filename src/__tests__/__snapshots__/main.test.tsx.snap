// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activities component activities snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-l8c9ks"
      >
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1o5btma"
            type="button"
          >
            2
             items left
          </button>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-0"
        >
          <div
            class="chakra-tabs css-0"
          >
            <div
              aria-orientation="horizontal"
              class="chakra-tabs__tablist css-1utx3w7"
              role="tablist"
            >
              <button
                aria-controls="tabs-:r8:--tabpanel-0"
                aria-disabled="false"
                aria-selected="true"
                class="chakra-tabs__tab css-an0qjo"
                data-index="0"
                id="tabs-:r8:--tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                All
              </button>
              <button
                aria-controls="tabs-:r8:--tabpanel-1"
                aria-disabled="false"
                aria-selected="false"
                class="chakra-tabs__tab css-an0qjo"
                data-index="1"
                id="tabs-:r8:--tab-1"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Active
              </button>
              <button
                aria-controls="tabs-:r8:--tabpanel-2"
                aria-disabled="false"
                aria-selected="false"
                class="chakra-tabs__tab css-an0qjo"
                data-index="2"
                id="tabs-:r8:--tab-2"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Completed
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1o5btma"
            type="button"
          >
            Clear completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-l8c9ks"
    >
      <div
        class="css-0"
      >
        <button
          class="chakra-button css-1o5btma"
          type="button"
        >
          2
           items left
        </button>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-0"
      >
        <div
          class="chakra-tabs css-0"
        >
          <div
            aria-orientation="horizontal"
            class="chakra-tabs__tablist css-1utx3w7"
            role="tablist"
          >
            <button
              aria-controls="tabs-:r8:--tabpanel-0"
              aria-disabled="false"
              aria-selected="true"
              class="chakra-tabs__tab css-an0qjo"
              data-index="0"
              id="tabs-:r8:--tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              All
            </button>
            <button
              aria-controls="tabs-:r8:--tabpanel-1"
              aria-disabled="false"
              aria-selected="false"
              class="chakra-tabs__tab css-an0qjo"
              data-index="1"
              id="tabs-:r8:--tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Active
            </button>
            <button
              aria-controls="tabs-:r8:--tabpanel-2"
              aria-disabled="false"
              aria-selected="false"
              class="chakra-tabs__tab css-an0qjo"
              data-index="2"
              id="tabs-:r8:--tab-2"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Completed
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-0"
      >
        <button
          class="chakra-button css-1o5btma"
          type="button"
        >
          Clear completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Activities component list snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-container css-1ig5rgb"
      >
        <div
          class="css-x0g6r9"
        >
          <div
            class="chakra-input__group css-bx0blc"
            data-group="true"
          >
            <div
              class="chakra-input__left-element css-1u2k0bl"
            >
              <svg
                class="chakra-icon css-1if5gp8"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              class="chakra-input css-1u3thfl"
              placeholder="What needs to be done?"
              value=""
            />
          </div>
        </div>
        <div
          class="chakra-stack css-zdx2uo"
        >
          <div
            class="css-1hfixyc"
          >
            <div
              class="chakra-stack css-t8kiha"
            >
              <label
                class="css-4g6ai3"
              >
                <input
                  hidden=""
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <div
                  aria-hidden="true"
                  class="css-py28wa"
                />
              </label>
              <p
                class="chakra-text css-1xph9jp"
              >
                Тестовое задание
              </p>
            </div>
          </div>
          <div
            class="css-1hfixyc"
          >
            <div
              class="chakra-stack css-t8kiha"
            >
              <label
                class="css-4g6ai3"
              >
                <input
                  checked=""
                  hidden=""
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <div
                  aria-hidden="true"
                  class="css-py28wa"
                  data-checked=""
                >
                  <svg
                    class="chakra-icon css-t1gj7g"
                    focusable="false"
                    viewBox="0 0 14 14"
                  >
                    <g
                      fill="currentColor"
                    >
                      <polygon
                        points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                      />
                    </g>
                  </svg>
                </div>
              </label>
              <s
                class="chakra-text css-1hitjv9"
              >
                Прекрасный код
              </s>
            </div>
          </div>
          <div
            class="css-1hfixyc"
          >
            <div
              class="chakra-stack css-t8kiha"
            >
              <label
                class="css-4g6ai3"
              >
                <input
                  hidden=""
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <div
                  aria-hidden="true"
                  class="css-py28wa"
                />
              </label>
              <p
                class="chakra-text css-1xph9jp"
              >
                Покрытие тестами
              </p>
            </div>
          </div>
        </div>
        <div
          class="css-l8c9ks"
        >
          <div
            class="css-0"
          >
            <button
              class="chakra-button css-1o5btma"
              type="button"
            >
              2
               items left
            </button>
          </div>
          <div
            class="css-17xejub"
          />
          <div
            class="css-0"
          >
            <div
              class="chakra-tabs css-0"
            >
              <div
                aria-orientation="horizontal"
                class="chakra-tabs__tablist css-1utx3w7"
                role="tablist"
              >
                <button
                  aria-controls="tabs-:r1:--tabpanel-0"
                  aria-disabled="false"
                  aria-selected="true"
                  class="chakra-tabs__tab css-an0qjo"
                  data-index="0"
                  id="tabs-:r1:--tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  All
                </button>
                <button
                  aria-controls="tabs-:r1:--tabpanel-1"
                  aria-disabled="false"
                  aria-selected="false"
                  class="chakra-tabs__tab css-an0qjo"
                  data-index="1"
                  id="tabs-:r1:--tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Active
                </button>
                <button
                  aria-controls="tabs-:r1:--tabpanel-2"
                  aria-disabled="false"
                  aria-selected="false"
                  class="chakra-tabs__tab css-an0qjo"
                  data-index="2"
                  id="tabs-:r1:--tab-2"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Completed
                </button>
              </div>
            </div>
          </div>
          <div
            class="css-17xejub"
          />
          <div
            class="css-0"
          >
            <button
              class="chakra-button css-1o5btma"
              type="button"
            >
              Clear completed
            </button>
          </div>
        </div>
        <div
          class="css-qvz413"
        />
        <div
          class="css-w8akdn"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-container css-1ig5rgb"
    >
      <div
        class="css-x0g6r9"
      >
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <div
            class="chakra-input__left-element css-1u2k0bl"
          >
            <svg
              class="chakra-icon css-1if5gp8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
          <input
            class="chakra-input css-1u3thfl"
            placeholder="What needs to be done?"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-stack css-zdx2uo"
      >
        <div
          class="css-1hfixyc"
        >
          <div
            class="chakra-stack css-t8kiha"
          >
            <label
              class="css-4g6ai3"
            >
              <input
                hidden=""
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <div
                aria-hidden="true"
                class="css-py28wa"
              />
            </label>
            <p
              class="chakra-text css-1xph9jp"
            >
              Тестовое задание
            </p>
          </div>
        </div>
        <div
          class="css-1hfixyc"
        >
          <div
            class="chakra-stack css-t8kiha"
          >
            <label
              class="css-4g6ai3"
            >
              <input
                checked=""
                hidden=""
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <div
                aria-hidden="true"
                class="css-py28wa"
                data-checked=""
              >
                <svg
                  class="chakra-icon css-t1gj7g"
                  focusable="false"
                  viewBox="0 0 14 14"
                >
                  <g
                    fill="currentColor"
                  >
                    <polygon
                      points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                    />
                  </g>
                </svg>
              </div>
            </label>
            <s
              class="chakra-text css-1hitjv9"
            >
              Прекрасный код
            </s>
          </div>
        </div>
        <div
          class="css-1hfixyc"
        >
          <div
            class="chakra-stack css-t8kiha"
          >
            <label
              class="css-4g6ai3"
            >
              <input
                hidden=""
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <div
                aria-hidden="true"
                class="css-py28wa"
              />
            </label>
            <p
              class="chakra-text css-1xph9jp"
            >
              Покрытие тестами
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-l8c9ks"
      >
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1o5btma"
            type="button"
          >
            2
             items left
          </button>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-0"
        >
          <div
            class="chakra-tabs css-0"
          >
            <div
              aria-orientation="horizontal"
              class="chakra-tabs__tablist css-1utx3w7"
              role="tablist"
            >
              <button
                aria-controls="tabs-:r1:--tabpanel-0"
                aria-disabled="false"
                aria-selected="true"
                class="chakra-tabs__tab css-an0qjo"
                data-index="0"
                id="tabs-:r1:--tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                All
              </button>
              <button
                aria-controls="tabs-:r1:--tabpanel-1"
                aria-disabled="false"
                aria-selected="false"
                class="chakra-tabs__tab css-an0qjo"
                data-index="1"
                id="tabs-:r1:--tab-1"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Active
              </button>
              <button
                aria-controls="tabs-:r1:--tabpanel-2"
                aria-disabled="false"
                aria-selected="false"
                class="chakra-tabs__tab css-an0qjo"
                data-index="2"
                id="tabs-:r1:--tab-2"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Completed
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1o5btma"
            type="button"
          >
            Clear completed
          </button>
        </div>
      </div>
      <div
        class="css-qvz413"
      />
      <div
        class="css-w8akdn"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
